### PR TITLE
Rz maps

### PR DESCRIPTION
**Pull request by zhut19**
**What is the problem / what does the code in this PR do**
 - Implement field-dependent radial-depth maps including `survival_probability_map`, `drift_speed_map`, `diffusion_longitudinal_map`
 - Two small fixes for scintillation timing, its dtype and moving the single electron center time to zero.

### CODE REVIEW

File: wfsim/core.py
Comment (ftoschi) [inline wfsim/core.py:647]: I am trying to follow the code here: E0 should be the constant electric field where the anode contribution is not dominating, dr the z step and rr the inverse of the position (constant if outside the anode-dominated field). If in the constant field region, isn't rr changing the value of the field, as it would be clipped to 1 / rA?

File: wfsim/core.py
Comment (ftoschi) [inline wfsim/core.py:652]: Why do remove the yield-weighted average time? This yields to negative time, is this supposed to happen? Also this average accounts for values outside the physical path of the electron in case of warping (as you take the larges gas gap for r)

File: wfsim/core.py
Comment (zhut19) [reply wfsim/core.py:647]: I don't remember doing this, but during the copy from fax to here, I merged a factor of rA into E0. So now this E0 equals to E (constant between rL and rA) * rA. And thus E0 * rr at inverse rA is that constant.

File: wfsim/core.py
Comment (zhut19) [reply wfsim/core.py:652]: Yeah, we do need this time shift, unfortunately, this is because the gate time we measure from data (the time shift applied to all S2s) is using the S2 center time. If we don't shift SE by its center time here, all the S2 center time will be reproduced a bit late. We will see Z_reconstructed deeper than Z_instruction.

File: wfsim/core.py
Comment (ftoschi) [reply wfsim/core.py:647]: Ah ok, then it makes sense

**Review by petergaemers — APPROVED** [review]
Hi @zhut19 is this finished or is there still something missing?

**Comment by zhut19** [comment]
> Hi @zhut19 is this finished or is there still something missing?

No, was just waiting for a bit of approval.

**Pull request merged**
